### ISSUE TITLE
QuSpin Installation on Macbook M1 Chip

### ISSUE THREAD

**Issue by JiahaoYao** [QuSpin/QuSpin#631]
There have been instances of users unable to install QuSpin on Macbook's M1 chip, as reported in the following GitHub issues: [#499](https://github.com/QuSpin/QuSpin/issues/499) and [#624](https://github.com/QuSpin/QuSpin/issues/624).

To overcome this problem, here is a step-by-step guide to installing QuSpin:

## Step 1: Install `mini-forge`

First, install `mini-forge` using the following command:

```bash
brew install miniforge
conda init zsh
```

## Step 2: Set Up a Conda osx-64 Environment on ARM Mac
Next, set up a conda osx-64 environment on your ARM Mac with the commands:

```bash
CONDA_SUBDIR=osx-64 conda create -n spin python=3.7
conda activate spin
conda config --env --set subdir osx-64
```

## Step 3: Install QuSpin
Finally, install QuSpin using the command:

```bash
conda install -c weinbe58 quspin
```
After following these steps, QuSpin should be successfully installed on your Macbook with the M1 chip. Enjoy using QuSpin!

**Comment by mgbukov** [comment]
thanks, @JiahaoYao 

Do you know if this installation makes use of the M1 chip features, or it simply allows you to run quspin on M1 devices?

**Comment by JiahaoYao** [comment]
Good question!

As far as I know, the installation process does not specifically exploit the features of the M1 chip. Instead, it sets up a conda environment in a manner similar to how it would function on an Intel chip, thus enabling the operation of QuSpin on M1 devices.

**Comment by AlexSchuckert** [comment]
Somehow this didn't work for me, it coulnd't solve the environment. Compiling it from scratch still worked well for me though if assigning the correct cython version:

`conda create -n MY_ENV python pybind11 cython=0.29.37 scipy numpy "blas=*=*accelerate*" joblib six dill gmpy2 numexpr boost llvm-openmp numba openmpi-mpicc`

By using "blas=*=*accelerate*" when installing numpy, the accelerate library is used which speeds up numpy operations by a factor of 2-4.

After this, install QuSpin by
`Git clone https://github.com/weinbe58/QuSpin.git
cd Quspin
python setup.py install --omp --record install_file.txt`

**Comment by mgbukov** [comment]
this should be fixed with quspin 1.0.0 (see doc on how to install)
